### PR TITLE
[[ Bug 19783 ]] Clear errors at end of try

### PR DESCRIPTION
**Pull request by montegoulding**
At the end of a try control structure any errors on the error
stack should be cleared. This was not happening if an error was
added to the error stack from some other context. Until we have
more context sensitive try locks and error stacks we should do
our best to clear any unthrown errors on the stack.

### CODE REVIEW

**Comment by runrevmark** [comment]
@montegoulding Is there a bug reported about improving the way 'try' works?

@livecode-vulcan review ok f72fa88

**Comment by livecode-vulcan** [comment]
💙 review by @runrevmark ok f72fa884cf5f94670f16a277296cd706f3c73bae

File: docs/notes/bugfix-19783.md [hunk 1]
```diff
@@ -0,0 +1 @@
+# Ensure any errors are cleared from teh error stack after a try control structure
```
Comment (livecodepanos) [inline docs/notes/bugfix-19783.md]: There is a typo here (`teh`) :)

**Comment by livecode-vulcan** [comment]
😎 test success f72fa884cf5f94670f16a277296cd706f3c73bae

* try-community-armv6-android-api8: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-api8/builds/336)
* try-community-armv6-android-api9: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-api9/builds/342)
* try-community-js-emscripten-sdk1.35: [success](https://vulcan.livecode.com/builders/try-community-js-emscripten-sdk1.35/builds/322)
* try-community-universal-ios-iphoneos10.3: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphoneos10.3/builds/265)
* try-community-universal-ios-iphonesimulator10.3: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphonesimulator10.3/builds/265)
* try-community-universal-mac-macosx10.6: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.6/builds/331)
* try-community-universal-mac-macosx10.9: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.9/builds/336)
* try-community-x86-linux-debian7: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian7/builds/334)
* try-community-x86-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian8/builds/340)
* try-community-x86_64-linux-debian7: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian7/builds/334)
* try-community-x86_64-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian8/builds/340)
* try-community-x86-win32: [success](https://vulcan.livecode.com/builders/try-community-x86-win32/builds/359)
* try-community-x86_64-win32: [success](https://vulcan.livecode.com/builders/try-community-x86_64-win32/builds/330)

**Comment by montegoulding** [comment]
@runrevmark not yet. Now that this has been accepted I'll move the rest of bug 19783 to a moomin report. If this wasn't accepted I would have turned 19783 into a moomin ;-)

**Comment by montegoulding** [comment]
http://quality.livecode.com/show_bug.cgi?id=19805

**Comment by montegoulding** [comment]
@runrevmark I will need to drop 4daafa3 and related tests from this patch as it adds a feature. I would be pleased to have general feedback before I do anything else here though. Can you think of anything else that needs testing?

**Comment by runrevmark** [comment]
I suggest splitting this up as follows (assuming it is possible):

- The patch which fixes the issue we encountered in the test runner (develop)
- The patch which fixes the docs about try being optional (develop-8.1)
- The patch which has the tests, with the tests which are broken marked as TestBroked (develop) - this means we can then iterate on them to make sure they represent the functionality we want (as there are few edge cases with regards 'return' and 'pass <handler>' as you've pointed out).
- A (WIP) patch which fixes the tests as current written.

If we then decide that some of the changes are too deep for 9 we can hold over until the next version.

**Comment by montegoulding** [comment]
OK, sure @runrevmark will do
